packages/processor/js/src/service/style/style.service.ts

```diff
@@ -359,6 +362,29 @@ function createStyleEntry(mapping: ThemeMappingType,
   return [key, value];
 }
 
+/**
+ * Tells if component requires all variant groups combinations to be generated.
+ * Basically that just means that default variant group values are not defined for all variant groups.
+ *
+ * @param mapping: ThemeMappingType - theme mapping configuration
+ * @param component: string - component name
+ *
+ * @return (boolean) - key identical to some of `source` keys if presents
+ */
+export function needsAllVariantCases(mapping: ThemeMappingType, component: string): boolean {
```

Comment:
Since the returned value defines further processing, I'd say this function should have it's own tests, expecting true/false for given requirements